internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -177,17 +191,20 @@ func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkf
 	dwctx, done := workflow.NewDisconnectedContext(wctx)
 	defer done()
 
-	sessionDurationHistogram.Record(
-		metricsCtx,
-		duration.Milliseconds(),
-		metric.WithAttributes(
-			attribute.Bool("replay", isReplaying),
-			attribute.Bool("success", err == nil),
-		),
-	)
+	sessionDurationHistogram.Record(metricsCtx, duration.Milliseconds(),
+		metric.WithAttributes(attribute.Bool("replay", isReplaying), attribute.Bool("success", err == nil)))
 
 	l = l.With(zap.Duration("duration", duration))
 
+	if eventTime != (time.Time{}) {
+		invocationDelay := time.Since(eventTime)
+		l = l.With(zap.Duration("invocation_delay", invocationDelay))
```

Comment:
why do we need to add it to every log message ?
or even log it at all?
maybe add some configurable threshold to log warn high latency if above threshold

Reply:
there will be only 1 log message with duration (see 197) and invocation delay

Reply:
but you replace l with a new l with the addition of invocation_delay

Reply:
I'm adding invocation_delay to the single log message that would be printed, yes.
previously this message contained duraiton only. Now it will contain invocation delay as well.
This is exactly what I wanted to achieve - 1 message (done/failed/etc) with duration and invocation_delay fields.
You want it to be done differently?

Reply:
I agree with @ig-ra on this. Adding (relevant) attributes to a log message doesn't reduce it's readability.

Reply:
I didn't say it reduces it's readability 
I meant I just added more attributes to all logs later instead of one specific logs

but I'm ok with it, resolving